tests/test_schema.py

```diff
@@ -4817,3 +4817,61 @@ def test_describe_escape(self):
             r"function test::foo() ->  std::str "
             r"using (SELECT r'\1');"
         )
+
+    def test_describe_poly_01(self):
+        self._assert_describe(
+            """
+            scalar type all extending str;
+
+            type Object {
+                property real -> bool;
+            }
+
+            function all() -> bool {
+                using (
+                    SELECT true
+                );
+            }
+            """,
+
+            'DESCRIBE OBJECT all AS TEXT',
+
+            """
+            function test::all() -> std::bool using (SELECT
+                true
+            );
+            scalar type test::all extending std::str;
```

Comment:
@elprans Do we intentionally support functions and types having the same names?

Reply:
IIRC we did at some point of time to allow converter functions share the same name as types they convert to, but then we adopted the `to_<typename>` naming and planned to remove this "feature".

Reply:
This may be worthwhile to discuss alongside the #1439 since it also touches on names uniqueness and how are functions different (or not) from other entities. Functions are somewhat special, but are they special enough is a good question.

Reply:
We don't "intentionally" allow this, there's just no name clash, because all function names are mangled.

Reply:
Well, I explained my -1 in #1439 and I think that name clashes should be prohibited.

Reply:
My understanding was that at the moment we make `\d` actually describe the status quo, as opposed to change behavior. At the moment the functions don't name clash.

Reply:
> At the moment the functions don't name clash.

Hm, looks like they do... you have `test::all` type and `test::all()` function, no?

But you're right, this PR should work withing the semantics implemented in the master branch. It looks OK to me, but I'll let @elprans to review it. The `for aliases in search_ns:` still looks questionable to me.

Reply:
When I said "they don't name clash" I meant that function and object by same name doesn't cause an error at the moment. But yeah, I am specifically testing cases where multiple entities have the same short name in the test, including a function and a a scalar.